codalab/apps/web/templates/web/index.html

```diff
@@ -47,7 +47,11 @@
                 <div class="col-sm-6 worksheet-random-list">
                     <div class="btn-group btn-group-justified" role="group" aria-label="...">
                           <div class="btn-group" role="group">
+                            {% if user.is_authenticated %}
                                 <a role="button" class="btn" href="/worksheets/?name=/">My Worksheet</a>
+                            {% else %}
+                                <a role="button" class="btn" href="/accounts/login/?next=/worksheets/?name=/">My Worksheet (Sign In First)</a>
+                            {% endif %}
```

Comment:
I think "(Sign in First)" probably not necessary if it takes you to login page anyway.